### PR TITLE
Fix problems in istio-proxy-reset job

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**
Improve istio-proxy-reset-job.
This PR addressed two problems:
- "null" parent for plain old Pods without backing deployment/replicaset etc.
- Poor Pod selection: Hard-coded old version of Istio Proxy to select the Pods that should be migrated. Logic is reversed now: the user provides the expected Istio Proxy version, and the logic selects all Pods that don't match this version.

Changes proposed in this pull request:

- Improved Pod matching logic
- Fixed problem with null parent
- Put the script into a separate file to support manual testing/maintenance
- Add dry-run option to support manual testing/maintenance
- Expose job's parameters as entries in values.yaml to simplify future maintenance

### CODE REVIEW

File: resources/istio/templates/istio-proxy-reset.yaml [hunk 1]
```diff
@@ -56,61 +56,14 @@ spec:
       containers:
         - name: proxy-reset
           image: eu.gcr.io/kyma-project/incubator/develop/k8s-tools:20201023-5de446cf
+          env:
+          - name: EXPECTED_ISTIO_PROXY_IMAGE
+            value: {{ .Values.kyma.proxyResetJob.expectedIstioProxyImage }}
```
Comment (Demonsthere) [inline resources/istio/templates/istio-proxy-reset.yaml:61]:
```suggestion
            value: {{ .Values.kyma.proxyResetJob.commonIstioProxyImagePrefix }}:{{ .Chart.Version }}
```

**Review by Demonsthere — DISMISSED** [review]
/meow

**Review by strekm — APPROVED** [review]
/meow

**Pull request merged**